sys/pm_callouts.c

```diff
@@ -885,8 +786,232 @@ void classifyAll(portmaster_packet_info* packetInfo,
             }
         }
 
+    }
+    return FWP_ACTION_NONE;
+}
+
+void classifyMultiple(
+    portmaster_packet_info* packetInfo,
+    verdict_cache_t* verdictCache,
+    KSPIN_LOCK* verdictCacheLock,
+    const FWPS_INCOMING_METADATA_VALUES* inMetaValues,
+    void* layerData,
+    FWPS_CLASSIFY_OUT* classifyOut
+    ) {
+    PNET_BUFFER_LIST nbl;
+    PNET_BUFFER nb;
+    UINT32 ipHeaderSize;
+    HANDLE handle;
+
+    /*
+     * The classifyFn may receive multiple netbuffer lists (chained), which in turn may each have multiple netbuffers.
+     * 
+     * Multiple netbuffer lists are possible in stream and forward layers.
+     * Multiple buffers are possible for outgoing data.
+     * 
+     * Source: https://docs.microsoft.com/en-us/windows-hardware/drivers/network/packet-indication-format
+     */
+
+    // Define variables.
+    FWPS_PACKET_INJECTION_STATE injection_state;
+
+    // First, run checks and get data that applies to all packets.
+
+    // sanity check
+    if (!packetInfo || !verdictCache || !verdictCacheLock || !inMetaValues || !layerData) {
+        ERR("Invalid parameters");
+        classifyOut->actionType = FWP_ACTION_BLOCK;
+        return;
+    }
```

Comment:
Since we're doing sanity checks here, we should probably also check `!classifyOut`